crates/frontend/src/components/monaco_editor.rs

```diff
@@ -37,22 +38,32 @@ pub fn monaco_editor(
             minimap_settings.set_enabled(Some(false));
             editor_settings.set_language(Some(language.to_string().as_str()));
             editor_settings.set_automatic_layout(Some(true));
-            editor_settings.set_value(Some(data_rs.get().as_str()));
+            editor_settings.set_font_family(Some("ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, \"Liberation Mono\", \"Courier New\", monospace"));
+            logging::log!("Monaco Init value {}", data);
+            editor_settings.set_value(Some(data.as_str()));
             editor_settings.set_render_final_newline(Some(true));
             editor_settings.set_read_only(Some(read_only));
             editor_settings.set_minimap(Some(&minimap_settings));
             let editor = CodeEditor::create(&node, Some(editor_settings));
-            editor_ws.update(|prev| {
-                prev.replace(Some(editor));
-            });
+
+            editor_ws.set(Rc::new(Some(editor)));
         }
     });
     view! {
-        <div id={node_id} class={styling} node_ref=editor_ref on:keyup=move |event| {
-            let new_data = event_target_value(&event);
-            logging::log!("Updating code");
-            data_ws.set_untracked(new_data);
-        }>
+        <div
+            id=node_id
+            class=styling
+            node_ref=editor_ref
+            on:keyup=move |_| {
+                match editor_rs.get().borrow() {
+                    Some(editor) => {
+                        logging::log!("here editor {:?}", editor.get_model().unwrap().get_value());
+                        on_change.call(editor.get_model().unwrap().get_value());
```

Comment:
Can we handle it gracefully? If we don't get the value , we should not call on_change . 
Also , we should avoid using plain unwrap